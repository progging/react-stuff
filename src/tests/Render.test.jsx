import React from "react"
import {Render} from ".."
import {mount} from 'enzyme'
import {mountToJson} from 'enzyme-to-json'

test("Children should not be rendered on falsy condition", () => {

    const component = mount(
        <Render if={false}>
            <div/>
        </Render>
    )

    const componentJson = mountToJson(component)
    expect(componentJson).toMatchSnapshot()
    expect(componentJson.children).toBe(null)
})

test("Children should be rendered on truthy condition", () => {

    const component = mount(
        <Render if={true}>
            <div/>
        </Render>
    )

    const componentJson = mountToJson(component)
    expect(componentJson).toMatchSnapshot()
    expect(componentJson.children).not.toBe(null)
})