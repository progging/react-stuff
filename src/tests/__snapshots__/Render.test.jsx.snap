// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Children should be rendered on truthy condition 1`] = `
<Render
  if={true}
>
  <div />
</Render>
`;

exports[`Children should not be rendered on falsy condition 1`] = `
<Render
  if={false}
/>
`;
